src/Support/Type.php

```diff
@@ -39,7 +39,7 @@ public function attributes(): array
     }
 
     /**
-     * @return array<string,array|string|FieldDefinition|Field>
+     * @return array<int|string,array|string|FieldDefinition|Field>
```

Comment:
Are you sure? Even in the linked readme example I can't see this. They "key" of fields always has to be a string, i.e. the name / identifier of the field.

Reply:
![image](https://user-images.githubusercontent.com/3076578/69638380-dc755700-105a-11ea-8127-e6ab6120ba4a.png)


these have int as key, since when you do not add a key it will be int

Reply:
O_O

Reply:
True, in this case, this part in `\Rebing\GraphQL\Support\Type::getFields` "makes it work":
```php
            } elseif ($field instanceof FieldDefinition) {
                $allFields[$field->name] = $field;
            } else {
```

Confusing

Reply:
Not sure how to each phpstan properly about this or if we should just ignore the errors.
```
 ------ ---------------------------------------------------------------------------------------------------------------- 
  Line   src/Support/Type.php                                                                                            
 ------ ---------------------------------------------------------------------------------------------------------------- 
  93     Property Rebing\GraphQL\Support\Field::$name (string) does not accept int|string.                               
  98     Parameter #1 $name of method Rebing\GraphQL\Support\Type::getFieldResolver() expects string, int|string given.  
 ------ ---------------------------------------------------------------------------------------------------------------- 
```
The static analyzer can't interfere the context in which we implicitly know that `$name` is an int 🤷‍♀️

Reply:
I wonder if it's possible to (re)write the `getFields` method to make this clearer so we don't need to nudge phpstan.

OTOH currently this method looks compact and not-so-bad to read, so we might opt for ignoring the errors. I'm undecided.

Either way, willing to merge this but tests need to be green anyway.

Reply:
Phpstan is actually correct in it assumptions because the typehint is still not correct.

since the int key in the return type should only apply when fieldDefinition is there. But don't think that is possible in docblock

I am ignoring the errors